src/components/guestbook/Guestbook.jsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import useEntry from '../../hooks/useEntry';
+import useUser from '../../hooks/useUser';
+
+function Guestbook() {
+  const [guestEntry, setGuestEntry] = useState('');
+  const { entry, setEntry } = useEntry();
+  const { user, setUser } = useUser();
+  const history = useHistory();
+
+  function updateGuestName() {
+    if (!guestEntry) {
+      alert('Please write an entry!');
+    } else {
+      setEntry([...entry, { name, message: guestEntry }]);
```

Comment:
This isn't actually updating the user's name currently. You'll want to set this to reference the stored user value that you get from your useAuth and useForm hooks.